runtime/centrifuge/src/migrations.rs

```diff
@@ -9,5 +9,163 @@
 // but WITHOUT ANY WARRANTY; without even the implied warranty of
 // MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 // GNU General Public License for more details.
+use cfg_primitives::Balance;
+use cfg_types::tokens::CurrencyId;
+use frame_support::{traits::OnRuntimeUpgrade, weights::Weight};
 
-pub type UpgradeCentrifuge1020 = ();
+use crate::Runtime;
+
+pub type UpgradeCentrifuge1020 = (asset_registry::CrossChainTransferabilityMigration,);
+
+mod asset_registry {
+	use cfg_types::{tokens as v1, tokens::CustomMetadata};
+	#[cfg(feature = "try-runtime")]
+	use frame_support::ensure;
+	use frame_support::{pallet_prelude::OptionQuery, storage_alias, Twox64Concat};
+	use orml_traits::asset_registry::AssetMetadata;
+	#[cfg(feature = "try-runtime")]
+	use sp_std::vec::Vec;
+
+	use super::*;
+	use crate::VERSION;
+
+	/// Migrate all the registered asset's metadata to the new version of
+	/// `CustomMetadata` which contains a `CrossChainTransferability` property.
+	/// At this point in time, the `transferability` of Tranche tokens should be
+	/// set to `CrossChainTransferability::Xcm` and for all other tokens to
+	/// `CrossChainTransferability::Xcm`, with the exception of
+	/// `Currency::Staking` tokens which are not registered in the first place.
+	pub struct CrossChainTransferabilityMigration;
+
+	// The old orml_asset_registry Metadata storage using v0::CustomMetadata
+	#[storage_alias]
+	type Metadata<T: orml_asset_registry::Config> = StorageMap<
+		orml_asset_registry::Pallet<T>,
+		Twox64Concat,
+		CurrencyId,
+		AssetMetadata<Balance, v0::CustomMetadata>,
+		OptionQuery,
+	>;
+
+	impl OnRuntimeUpgrade for CrossChainTransferabilityMigration {
+		fn on_runtime_upgrade() -> Weight {
+			// todo(nuno): this migration will **most likely** be part of the set of
+			// migrations when upgrading to Centrifuge 1020. Refine when we know
+			// for sure.
+			if VERSION.spec_version != 1020 {
+				return Weight::zero();
+			}
+
+			orml_asset_registry::Metadata::<Runtime>::translate(
+				|asset_id: CurrencyId, old_metadata: AssetMetadata<Balance, v0::CustomMetadata>| {
+					match asset_id {
+						CurrencyId::Staking(_) => None,
+						CurrencyId::Tranche(_, _) => Some(to_metadata_v1(
+							old_metadata,
+							v1::CrossChainTransferability::Connectors,
+						)),
+						_ => Some(to_metadata_v1(
+							old_metadata.clone(),
+							v1::CrossChainTransferability::Xcm(old_metadata.additional.xcm),
+						)),
+					}
+				},
+			);
+
+			let n = orml_asset_registry::Metadata::<Runtime>::iter().count() as u64;
```

Comment:
I assume that the number of items is low, or empty, in Centrifuge right?

Reply:
It's low for sure :) just a few entries now